### PR TITLE
RavenDB-11954: Retry on EINVAL pwrite fail if cifs/nfs mount

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Sparrow/Platform/Posix/Syscall.cs [hunk 1]
```diff
@@ -275,12 +276,34 @@ public static int AllocateFileSpace(int fd, long size, string file, out bool usi
                         // fallocate is not supported, we'll use lseek instead
                         usingWrite = true;
                         byte b = 0;
-                        if (pwrite(fd, &b, 1, size - 1) != 1L)
+                        int cifsRetry = 3;
+                        while (true)
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/Syscall.cs]: You can't do a `while(true)`

File: src/Sparrow/Platform/Posix/Syscall.cs [hunk 2]
```diff
@@ -275,12 +276,34 @@ public static int AllocateFileSpace(int fd, long size, string file, out bool usi
...
+                                    // lets give it few retries with short pauses between them - RavenDB-11954
+                                    if (CheckSyncDirectoryAllowed(file) == false) // cifs/nfs
+                                    {
+                                        if (cifsRetry-- <= 0)
+                                        {
+                                            ThrowLastError(err,
+                                                "Failed to pwrite WITH retries (on cifs/nfs mount) in order to fallocate where fallocate is not supported for " + file);
+                                        }
+
+                                        Thread.Sleep(200);
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/Syscall.cs]: Can we do less than that? Maybe compensate with increasing the number of retries

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Sparrow/Platform/Posix/Syscall.cs (hunk 1 above)
Comment (aviviadi) [reply src/Sparrow/Platform/Posix/Syscall.cs]: I'll write `static int RetriesOrThrow()` for both cifs retries and EINTR w/wo timeouts

File: src/Sparrow/Platform/Posix/Syscall.cs (hunk 2 above)
Comment (aviviadi) [reply src/Sparrow/Platform/Posix/Syscall.cs]: Thread.Sleep(10 * (cifsRetry ^ 3)); // wait upto 3 time in intervals of : 10mSec, 80mSec, 270mSec

File: src/Sparrow/Platform/Posix/Syscall.cs (hunk 1 above)
Comment (ayende) [reply src/Sparrow/Platform/Posix/Syscall.cs]: ok

**Pull request merged**